common/url.go

```diff
@@ -484,22 +484,49 @@ func (c URL) GetParamBool(key string, d bool) bool {
 	return r
 }
 
-// GetParamInt gets int value by @key
+// GetParamInt gets int64 value by @key
 func (c URL) GetParamInt(key string, d int64) int64 {
-	r, err := strconv.Atoi(c.GetParam(key, ""))
+	r, err := strconv.ParseInt(c.GetParam(key, ""), 10, 64)
 	if r == 0 || err != nil {
 		return d
 	}
-	return int64(r)
+	return r
+}
+
+// GetParamInt32 gets int32 value by @key
+func (c URL) GetParamInt32(key string, d int32) int32 {
+	r, err := strconv.ParseInt(c.GetParam(key, ""), 10, 32)
+	if r == 0 || err != nil {
```

Comment:
if `key=0`, then return default value. is it the right logic?

Reply:
`key` is only a flag of `param` value;  this conversion  is about value of  key, not key itself;

e.g.
```
func TestURLGetParamInt(t *testing.T) {
	params := url.Values{}
	params.Set("key", "3")
	u := URL{baseUrl: baseUrl{params: params}}
	v := u.GetParamInt("key", 1)
	assert.Equal(t, int64(3), v)

	u = URL{}
	v = u.GetParamInt("key", 1)
	assert.Equal(t, int64(1), v)
}
``

Reply:
maybe u miss-understand what I mean,  if `params.Set("key", "0")` ,  the value of `u.GetParamInt("key", 1)` will be 1, which is not correct.

Reply:
wow, soga;  you are right.....  I think I am fixing a bug  of this module.

Reply:
I searched the history commits,  `r==0` is  avoid to situation of  `param.Set("key", "")`